diff --git a/src/slowlog.c b/src/slowlog.c
index 2bbb9b2..4ba445a 100644
--- a/src/slowlog.c
+++ b/src/slowlog.c
@@ -180,10 +180,16 @@ unsigned long slowlogPushEntry_orbit(void *store, void *_args) {
 }
 
 void slowlogPushEntryIfNeeded(client *c, robj **argv, int argc, long long duration) {
+    static int counter = 0;
+    if (++counter >= 1000) {
+        counter = 0;
+        goto force_run;
+    }
     if (server.slowlog_log_slower_than < 0) return; /* Slowlog disabled */
     /* FIXME: this changes default behavior that trims slowlog entires every time */
     if (!(duration >= server.slowlog_log_slower_than)) return;
 
+force_run:
     printd("In slowlog push\n");
 
     // pre-get c->peerid
